src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -703,6 +703,50 @@ public void onFileChange(File file) {
         return dockerfileLines;
     }
 
+    /**
+     * Trim all lines and get them without comments or empty lines.
+     */
+    protected static List<String> getCleanedLines(List<String> dockerfileLines) throws PluginExecutionException {
+        List<String> result = new ArrayList<String>();
+        for (String line : dockerfileLines) {
+            // Remove white space from the beginning and end of the line
+            String pendingLine = line.trim();
+            int commentIndex = pendingLine.indexOf("#");
+            if (commentIndex >= 0) {
+                String contentBeforeSymbol = pendingLine.substring(0, commentIndex);
+                // trim again after removing trailing comment
+                pendingLine = contentBeforeSymbol.trim();
+            }
+            if (!pendingLine.isEmpty()) {
+                result.add(pendingLine);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Combine multi-line commands into single lines
+     */
+    protected static List<String> getCombinedLines(List<String> dockerfileLines) throws PluginExecutionException {
+        List<String> result = new ArrayList<String>();
+        int i = 0;
+        while (i < dockerfileLines.size()) {
+            String pendingLine = dockerfileLines.get(i).trim();
+            int multilineIndex;
+            int j = i+1;
+            while ((multilineIndex = pendingLine.indexOf("\\")) >= 0 && j < dockerfileLines.size()) {
```

Comment:
Technically you can use \ in a filename on Linux. Can we document this method only handles "clean" lines (no comments or trailing whitespace) so that we can check for lines that end with '\\'?

Reply:
Fixed